pkg/controller/provider/validation.go

```diff
@@ -235,8 +238,34 @@ func (r *Reconciler) validateSecret(provider *api.Provider) (secret *core.Secret
 		}
 
 		url, _ := url.Parse(provider.Spec.URL)
-		if crt, err := util.GetTlsCertificate(url, secret); err == nil {
-			provider.Status.Fingerprint = util.Fingerprint(crt)
+		if cert, err := util.GetTlsCertificate(url, secret); err == nil {
+			provider.Status.Fingerprint = util.Fingerprint(cert)
+
+			tlsConfig := &tls.Config{
+				RootCAs: x509.NewCertPool(),
+			}
+
+			// Add the server certificate to the certificate pool
+			tlsConfig.RootCAs.AddCert(cert)
+
+			// We want to confirm that url.Host does not already include a port to avoid duplication
+			host := url.Host
```

Comment:
This uses the specified url, but if the user specifies the host, we resolve the hostname to the IP (added in your PR https://github.com/kubev2v/forklift/pull/1490). In these cases the certs can act differently
Could we get the address from `getSourceDetails`  or something similar?